src/views/Projects/index.jsx

```diff
@@ -20,11 +21,32 @@ import ProjectCard from '../../components/ProjectCard';
 export default class Projects extends Component {
   state = {
     projects,
+    search: '',
+  };
+  handleTextInputChange = event => {
+    this.setState({ search: event.target.value });
   };
 
   render() {
     const { classes } = this.props;
-    const { projects } = this.state;
+    const { projects, search } = this.state;
+    const selectedProjects = Object.keys(projects)
+      .filter(fileName => {
+        const found =
+          fileName.toLowerCase().indexOf(search.toLowerCase()) !== -1 ||
+          projects[fileName].name
+            .toLowerCase()
+            .indexOf(search.toLowerCase()) !== -1;
+
+        return found;
+      })
+      .reduce(
+        (prev, key) => ({
+          ...prev,
+          [key]: projects[key],
+        }),
+        {}
+      );
```

Comment:
We can replace all that logic with:

```js
const filteredProjects = Object.keys(projects)
  .filter(fileName =>
    projects[fileName].name.toLowerCase().includes(searchTerm.toLowerCase())
  )
  .reduce(
    (prev, key) => ({
      ...prev,
      [key]: projects[key],
    }),
    {}
  );
```

Reply:
Ok! wah!! i didnt know it could be this simple 💯